### PR TITLE
Add PT2E cv&llm example

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

example

## Description

- [x] cv static 
- [ ] cv dynamic
- [x] nlp static

Dynamic accuracy currently has issues, so once it's fixed, we will add the example.

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by chensuyue** [comment]
Please add the model config and run_quant.sh/run_benchmark.sh

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Step-by-Step
+============
+This document describes the step-by-step instructions to run large language models (LLMs) on 4th Gen Intel® Xeon® Scalable Processor (codenamed Sapphire Rapids) with PyTorch 2 Export Quantization.
+
+The script `run_clm_no_trainer.py` supports `OPT` quantization and validates last word prediction accuracy with [lm_eval](https://github.com/EleutherAI/lm-evaluation-harness.git) now, and we are adding more models.
```
Comment (yiliu30) [inline examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/README.md:5]:
```suggestion
Currently, users can use `run_clm_no_trainer.py` to quantize the `OPT` series models and validate the last word prediction accuracy with [lm_eval](https://github.com/EleutherAI/lm-evaluation-harness.git). We will add more models in the near future.
```

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
+import argparse
+import sys
+
+sys.path.append('./')
```
Comment (yiliu30) [inline examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py:4]:
```suggestion
```

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+if args.quantize:
+    
+    from neural_compressor.torch.quantization import (
+            convert,
+            get_default_static_config,
+            prepare,
+        )
+    from neural_compressor.torch.export import export
+    from torch.export import Dim
+    # set TOKENIZERS_PARALLELISM to false
```
Comment (yiliu30) [inline examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py:68]:
```suggestion
```

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    def get_example_inputs(tokenizer):
+        text = "Hello, welcome to LLM world."
+        encoded_input = tokenizer(text, return_tensors="pt")
+
+        example_inputs = encoded_input
+        # print(f"example_inputs: {example_inputs}")
```
Comment (yiliu30) [inline examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py:74]:
```suggestion
```

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        input_ids = example_inputs["input_ids"]
+        input_ids_batch = torch.cat((input_ids, input_ids), dim=0)
+        print(f"input_ids_batch shape: {input_ids_batch.shape}")
+        tuple_inputs = (input_ids_batch,)
+        return tuple_inputs
+    # os.environ["TOKENIZERS_PARALLELISM"] = "false"
```
Comment (yiliu30) [inline examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py:80]:
```suggestion
```

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+    dynamic_shapes = {"input_ids": (batch, seq_len)}
+    example_inputs = get_example_inputs(tokenizer)
+    exported_model = export(user_model, example_inputs=example_inputs, dynamic_shapes=dynamic_shapes)
+
+    quant_config = get_default_static_config()
+    # prepare
+    prepare_model = prepare(exported_model, quant_config)
+
+    # calibrate
+    for i in range(2):
```
Comment (yiliu30) [inline examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py:93]: Is 2 batches enough?

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    for task_name in args.tasks.split(","):
+        if task_name == "wikitext":
+            acc = results["results"][task_name]["word_perplexity,none"]
+        else:
+            acc = results["results"][task_name]["acc,none"]
+    print("Accuracy: %.5f" % acc)
+    print('Batch size = %d' % args.batch_size)
+
+if args.performance:
+    # user_model.eval()
```
Comment (yiliu30) [inline examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py:128]:
```suggestion
```

**Review by yiliu30 — CHANGES_REQUESTED** [review]

File: examples/3.x_api/pytorch/cv/static_quant/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# ImageNet Quantization
+
+This implements quantization of popular model architectures, such as ResNet, AlexNet, and VGG on the ImageNet dataset.
```
[the remark this answers is not in the document]
Comment (Kaihui-intel) [reply examples/3.x_api/pytorch/cv/static_quant/README.md]: fixed https://github.com/intel/neural-compressor/pull/1853/commits/e8d6888dbd82709248ab0c759a2c49109b968898

File: examples/3.x_api/pytorch/cv/static_quant/main.py [hunk 9]
```diff
@@ -0,0 +1,514 @@
...
+    val_loader = torch.utils.data.DataLoader(
+        val_dataset, batch_size=args.batch_size, shuffle=False,
+        num_workers=args.workers, pin_memory=True, sampler=val_sampler)
+    
+    if args.quantize:
+        from neural_compressor.torch.export import export
+        from neural_compressor.torch.quantization import prepare, convert, get_default_static_config
+
+        # Prepare the float model and example inputs for export model
+        model = model
```
[the remark this answers is not in the document]
Comment (Kaihui-intel) [reply examples/3.x_api/pytorch/cv/static_quant/main.py]: fixed https://github.com/intel/neural-compressor/pull/1853/commits/e8d6888dbd82709248ab0c759a2c49109b968898

File: examples/3.x_api/pytorch/cv/static_quant/main.py [hunk 10]
```diff
@@ -0,0 +1,514 @@
...
+        config.freezing = True
+        opt_model = torch.compile(q_model)
+        model = opt_model
+
+    
+    if args.evaluate:
+        validate(val_loader, model, criterion, args)
+        return
+
+def train(train_loader, model, criterion, optimizer, epoch, device, args):
```
[the remark this answers is not in the document]
Comment (Kaihui-intel) [reply examples/3.x_api/pytorch/cv/static_quant/main.py]: fixed https://github.com/intel/neural-compressor/pull/1853/commits/e8d6888dbd82709248ab0c759a2c49109b968898

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py (hunk 3 above)
Comment (Kaihui-intel) [reply examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py]: fixed https://github.com/intel/neural-compressor/pull/1853/commits/96c946fd79f3a9102c47d81d58d8ea5c87d08612

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py (hunk 5 above)
Comment (Kaihui-intel) [reply examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py]: fixed https://github.com/intel/neural-compressor/pull/1853/commits/96c946fd79f3a9102c47d81d58d8ea5c87d08612

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py (hunk 7 above)
Comment (Kaihui-intel) [reply examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py:124]: fixed https://github.com/intel/neural-compressor/pull/1853/commits/96c946fd79f3a9102c47d81d58d8ea5c87d08612

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py (hunk 4 above)
Comment (Kaihui-intel) [reply examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py]: https://github.com/intel/neural-compressor/pull/1853/commits/96c946fd79f3a9102c47d81d58d8ea5c87d08612

File: examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py (hunk 6 above)
Comment (Kaihui-intel) [reply examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py:90]: Verified that the accuracy remains unchanged as the batch size increases.

Comment (yiliu30) [reply examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/static_quant/pt2e/run_clm_no_trainer.py:90]: How about exposing an arg for use to select the number of iterations?

**Review by yiliu30 — APPROVED** [review]

**Review by chensuyue — APPROVED** [review]
